### PR TITLE
FT-04def: added image,JD,Teams field

### PR DESCRIPTION
**Pull request by Jeysh-git**
In this PR,
   - created Avatar component and added user profile image upload.
   - added bs-datepicker to select joining date
   - added ember-power-select-tool to select team list.
Figma Design Link:
https://www.figma.com/file/i9whcNdWDHjHsXVdpoYyFh/New-Joinee?node-id=1%3A2937

### CODE REVIEW

File: app/components/avatar-upload.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+           
+            preview.src = reader.result;
+            }, false);
+        
+            if (file) {
+              reader.readAsDataURL(file);
+            }
+
+            
+            this.get('member-profile').set('image',preview.src)
```
Comment (shenbaga-praburaj) [inline app/components/avatar-upload.js]: We're setting the image value before preview.src is being set. Check the `load` event listener.

File: app/components/user-form.js [hunk 2]
```diff
@@ -4,28 +4,36 @@ import EmberObject from '@ember/object';
...
+           //submit logic  
          },
+        
          reset(){
             // reset logic
         },
+
+        selectTeams(teamname){
+            
+            this.set('selectedoption',teamname)
```
Comment (shenbaga-praburaj) [inline app/components/user-form.js]: We receive the team object as param. But the name can be misleading.

File: app/styles/datepicker.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    width:300px;
+   // line-height: 30px;
+    font-weight: 300;
+    content:none;
+    
+
+
+}
+
+ tr,td,th {
```
Comment (shenbaga-praburaj) [inline app/styles/datepicker.scss]: tr, td are very common to be used all over the product. So, it's always recommended to write component specific css with proper classnames hierarchy. Kindly check all the CSS for the same.

File: app/styles/teamspowerselect.scss [hunk 4]
```diff
@@ -0,0 +1,60 @@
+// .ember-power-select-search ,input {
+
+// placeholder:"search";
+// }
+
+.ember-power-select-trigger{
+    width:300px;
+}
+
```
Comment (shenbaga-praburaj) [inline app/styles/teamspowerselect.scss:5]: For Information:
If we are going to apply this width for all power select used in our product, this declaration is fine. But if we have to be specific about some form, then we need to follow the class hierarchy.

File: app/templates/components/user-avatar.hbs [hunk 5]
```diff
@@ -4,7 +4,7 @@
     {{#if empImage}}
     <div class = "employee-image">
         {{svg-jar empImage height="80px"}}
```
Comment (shenbaga-praburaj) [inline app/templates/components/user-avatar.hbs]: We can go with. `` tag itself for images.

File: app/components/avatar-upload.js (hunk 1 above)
Comment (Jeysh-git) [reply app/components/avatar-upload.js]: Done. As discussed, setting of image value will be handled while saving the form data. So, I'm removing this code here.

File: app/styles/datepicker.scss (hunk 3 above)
Comment (Jeysh-git) [reply app/styles/datepicker.scss]: Done.

File: app/components/user-form.js (hunk 2 above)
Comment (Jeysh-git) [reply app/components/user-form.js]: Done.

File: app/components/user-form.js [hunk 6]
```diff
@@ -4,28 +4,37 @@ import EmberObject from '@ember/object';
 export default Component.extend({
     
     formbuttons:null,
+    selectedoption: '',
+    teams: '',
 
     init(){
         this._super(...arguments);
 
+        this.set('teams',['Freshteam1','Freshteam2','Freshteam3','Freshteam4'])
```
Comment (shenbaga-praburaj) [inline app/components/user-form.js]: We Should not define same array in different places. We should either define in a single place like in service, or we should get them from store/mirage/etc. Coz, if we change the list in future, we'll have to change in all the places.

**Review by shenbaga-praburaj — APPROVED** [review]

**Pull request merged**